### PR TITLE
Visual 2022

### PR DESCRIPTION
**Pull request by ArtyomBanchikov**

### CODE REVIEW

File: Pets.API/Controllers/PetController.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            var dbPet = await _context.Pets.FindAsync(request.Id);
+            if (dbPet == null)
+                return BadRequest("Pet not found");
+
+            dbPet.Age = request.Age;
+            dbPet.Weight = request.Weight;
+            dbPet.Name = request.Name;
+
+            _context.Entry(dbPet).State = EntityState.Modified;
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: Check method update

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Comment by hromau** [comment]
why name PR is so strange?

File: Pets.API/Controllers/PetController.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        private readonly PetsContext _context;
+
+        public PetController(PetsContext context)
+        {
+            _context = context; 
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Pet>>> GetAll()
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: delete all async-await key-words, ur application will work fast without it. delete actionresult, return only objects

File: Pets.API/Program.cs [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+string connection = builder.Configuration.GetConnectionString("DefaultConnection");
+
+// ��������� �������� ApplicationContext � �������� ������� � ����������
```
Comment (hromau) [inline Pets.API/Program.cs]: äîáàâëÿåì êîíòåêñò ApplicationContext â êà÷åñòâå ñåðâèñà â ïðèëîæåíèå

File: Pets.API/Properties/launchSettings.json [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
-      "launchUrl": "weatherforecast",
+      "launchUrl": "swagger",
+      "applicationUrl": "https://localhost:7139;http://localhost:5139",
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
-    "Pets.API": {
-      "commandName": "Project",
+    "IIS Express": {
```
Comment (hromau) [inline Pets.API/Properties/launchSettings.json]: u can delete iis expres setup, cause u willn't use it

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.API/Controllers/PetController.cs [hunk 5]
```diff
@@ -42,15 +42,7 @@ public void Post(Pet pet)
         [HttpPut]
         public void Put(Pet request)
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: Rename paramet
What is request?

File: Pets.API/Controllers/PetController.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        public void Post(Pet pet)
+        {
+            _context.Pets.Add(pet);
+            _context.SaveChanges();
+        }
+
+        [HttpPut]
+        public void Put(Pet request)
+        {
+            _context.Update(request);
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: _context.Pets.Update(request);

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Review by hromau — APPROVED** [review]

File: Pets.API/Controllers/PetController.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        public Pet Get(int id)
+        {
+            var pet = _context.Pets.Find(id);
+            if (pet == null)
+                throw new ArgumentNullException("Object cannot be null", nameof(pet));
+            return pet;
+        }
+
+        [HttpPost]
+        public void Post(Pet pet)
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: change

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.API/Controllers/PetController.cs (hunk 7 above)
Comment (ArtyomBanchikov) [reply Pets.API/Controllers/PetController.cs]: Change what?

File: Pets.API/Controllers/PetController.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        [HttpGet]
+        public IEnumerable<Pet> GetAll()
+        {
+            return _context.Pets.ToList();
+        }
+
+        [HttpGet("{id}")]
+        public Pet Get(int id)
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: Rename GetById

File: Pets.API/Controllers/PetController.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+        [HttpPost]
+        public void Add(Pet pet)
+        {
+            _context.Pets.Add(pet);
+            _context.SaveChanges();
+        }
+
+        [HttpPut]
+        public void Put(Pet pet)
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: Rename to Update

**Review by stas1408123 — APPROVED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**